crates/simulacrum/src/lib.rs

```diff
@@ -472,8 +473,8 @@ impl<T, V: store::SimulatorStore> ReadStore for Simulacrum<T, V> {
     fn get_transaction(
         &self,
         tx_digest: &sui_types::digests::TransactionDigest,
-    ) -> sui_types::storage::error::Result<Option<VerifiedTransaction>> {
-        Ok(self.store().get_transaction(tx_digest))
+    ) -> sui_types::storage::error::Result<Option<Arc<VerifiedTransaction>>> {
+        Ok(self.store().get_transaction(tx_digest).map(Arc::new))
```

Comment:
For change like this, does it worth verifying memory usage?

Reply:
well, as far as I know we aren't even using this crate right now. But, broadly speaking, this should reduce memory usage and CPU once we are servicing requests from the cache. More and more of these APIs will start to use Arc<T> so that we can return a cloned Arc from the cache rather than doing a deep Clone.